core/tests/ts-integration/tests/api/snapshots-creator.test.ts

```diff
@@ -0,0 +1,84 @@
+import { TestMaster } from '../../src/index';
+import fs from 'fs';
+import * as zlib from 'zlib';
+import { snapshots_creator } from 'zk/build/run/run';
+describe('Snapshots API tests', () => {
+    let testMaster: TestMaster;
+
+    beforeAll(() => {
+        testMaster = TestMaster.getInstance(__filename);
+
+        if (process.env.ZKSYNC_ENV!.startsWith('ext-node')) {
+            console.warn("You are trying to run snapshots creator tests on external node. It's not supported.");
+        }
+    });
+
+    async function runCreator() {
+        console.log('Starting creator');
+        await snapshots_creator();
+    }
+
+    async function rpcRequest(name: string, params: any) {
+        const response = await testMaster.mainAccount().provider.send(name, params);
+        console.log(response);
+        return response;
+    }
+
+    async function getAllSnapshots() {
+        return await rpcRequest('snapshots_getAllSnapshots', []);
+    }
+
+    async function getSnapshot(snapshotL1Batch: number) {
+        return rpcRequest('snapshots_getSnapshot', [snapshotL1Batch]);
+    }
+
+    async function decompressGzip(filePath: string): Promise<string> {
+        return new Promise((resolve, reject) => {
+            const readStream = fs.createReadStream(filePath);
+            const gunzip = zlib.createGunzip();
+            let data = '';
+
+            gunzip.on('data', (chunk) => (data += chunk.toString()));
+            gunzip.on('end', () => resolve(data));
+            gunzip.on('error', reject);
+
+            readStream.pipe(gunzip);
+        });
+    }
+    async function createAndValidateSnapshot() {
+        let existingBatchNumbers = (await getAllSnapshots()).snapshotsL1BatchNumbers as number[];
```

Comment:
Nit: Most of these `let`s could be replaced with `const`s since JS / TS const-ness is shallow, rather than structural as in Rust; if a variable itself is not reassigned, it can be a `const`; and at least some code styles for JS / TS consider `const`s preferable. This is obviously not a huge issue, just FYI.

Reply:
I got too used to IDE nudging me about such issues (I must either get WebStorm again or actively start checking such issues); corrected